File: app/src/main/java/sg/com/agentapp/api/api_clients/RetroOneMapAPIClient.java
//package sg.com.agentapp.api.api_clients;
//
///* Created by jayyong on 03/04/2018. */
//
//import retrofit2.Retrofit;
//import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
//import retrofit2.converter.gson.GsonConverterFactory;
//
//public class RetroOneMapAPIClient {
//    private static final String BASE_URL = "https://developers.onemap.sg/";
//    private static Retrofit retrofit = null;
//
//    private String a, b, c;
//
//    //    private final static OkHttpClient okHttpClient = new OkHttpClient.Builder()
////            .connectTimeout(20, TimeUnit.SECONDS)
////            .writeTimeout(20, TimeUnit.SECONDS)
////            .readTimeout(30, TimeUnit.SECONDS)
////            .build();
//    public synchronized static Retrofit getClient() {
//        if (retrofit == null) {
//            retrofit = new Retrofit.Builder()
//                    .baseUrl(BASE_URL)
//                    .addConverterFactory(GsonConverterFactory.create())
//                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
////                    .client(okHttpClient)
//                    .build();
//        }
//        return retrofit;
//    }
//
//}

File: app/src/main/java/sg/com/agentapp/api/api_clients/RetroOneMapAPIClient.kt
package sg.com.agentapp.api.api_clients

/* Created by jayyong on 16/10/2018. */

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import sg.com.agentapp.api.RetroOkClient
import sg.com.agentapp.api.interfaces.OneMapsApiInterface

object RetroOneMapAPIClient {
    val BASE_URL = "https://developers.onemap.sg/"

    private var retrofit: Retrofit
    private var apiInterface: OneMapsApiInterface

    init {
        retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(RetroOkClient().okHttpClient)
                .build()

        apiInterface = retrofit.create(OneMapsApiInterface::class.java)

    }


    val api: OneMapsApiInterface
        get() {
            return apiInterface
        }
}
